### ISSUE TITLE
Eslint lint error after "!.storybook" added

### ISSUE THREAD

**Issue by Rey-Wang** [storybookjs/eslint-plugin-storybook#154]
**Describe the bug**
After I added the `!.storybook` into the Eslint ignorePatterns, got the error that package.json path is not correct
Error: `The provided path in your eslintrc.json - /****/monorepo/package.json is not a valid path to a package.json file or your package.json file is not in the same folder as ESLint is running from.

**To Reproduce**
my folder structure
monorepo
    - web
    - package.json
    - .storybook
    - stories

**Expected behavior**
expect the eslint work properly

**Comment by andzno1** [comment]
I've got the same issue after running `storybook automigrate` in an nx project. The migration added `storybook.configs['flat/recommended']` to my ESLint configuration. Linting on the terminal works without any issue. Linting errors for files inside my `.storybook` directories are being displayed.

The ESLint VS Code extension seems to have a problem, though:
```text
2025-08-08 16:02:33.298 [error] An unexpected error occurred:
2025-08-08 16:02:33.298 [error] Error: Error: The provided path in your eslintrc.json - /home/node/workspace/project/libs/library-root/library-name/package.json is not a valid path to a package.json file or your package.json file is not in the same folder as ESLint is running from.

Read more at: https://github.com/storybookjs/storybook/blob/next/code/lib/eslint-plugin/docs/rules/no-uninstalled-addons.md
Occurred while linting /home/node/workspace/project/libs/library-root/library-name/.storybook/main.ts:20
Rule: "storybook/no-uninstalled-addons"
    at reportUninstalledAddons (/home/node/workspace/project/node_modules/eslint-plugin-storybook/dist/index.js:11:1142)
    at findAddonsPropAndReport (/home/node/workspace/project/node_modules/eslint-plugin-storybook/dist/index.js:11:1996)
    at ExportDefaultDeclaration (/home/node/workspace/project/node_modules/eslint-plugin-storybook/dist/index.js:11:2258)
    at ruleErrorHandler (/home/node/workspace/project/node_modules/eslint/lib/linter/linter.js:1310:33)
    at /home/node/workspace/project/node_modules/eslint/lib/linter/safe-emitter.js:45:46
    at Array.forEach (<anonymous>)
    at Object.emit (/home/node/workspace/project/node_modules/eslint/lib/linter/safe-emitter.js:45:26)
    at NodeEventGenerator.applySelector (/home/node/workspace/project/node_modules/eslint/lib/linter/node-event-generator.js:171:17)
    at NodeEventGenerator.applySelectors (/home/node/workspace/project/node_modules/eslint/lib/linter/node-event-generator.js:227:10)
    at NodeEventGenerator.enterNode (/home/node/workspace/project/node_modules/eslint/lib/linter/node-event-generator.js:241:8)
```
